.github/workflows/ci.yml

```diff
@@ -79,50 +100,42 @@ jobs:
       uses: actions/cache@v2
       with:
         path: |
-          ~/.cargo/registry
-          ~/.cargo/git
-        key: ${{ runner.os }}-cargo-pg12-cargo
-        restore-keys: |
-          ${{ runner.os }}-cargo-pg12-3
-          ${{ runner.os }}-cargo-pg12-2
-          ${{ runner.os }}-cargo-pg12
+          /usr/local/cargo/registry
+          /usr/local/cargo/git
+        key: TESTING6-${{ runner.os }}-test-pg${{ env.PGVERSION }}-cargo
 
     - name: Cache cargo target dir
       uses: actions/cache@v2
       with:
         path: target
-        key: ${{ runner.os }}-cargo-build-target-pg12-4
-        restore-keys: |
-          ${{ runner.os }}-cargo-build-target-pg12-3
-          ${{ runner.os }}-cargo-build-target-pg12-2
-          ${{ runner.os }}-cargo-build-target-pg12
+        key: TESTING6-${{ runner.os }}-test-pg${{ env.PGVERSION }}-target
 
-    - name: Run PG 12 Tests
-      run: sudo -HEsu postgres sh -c "/usr/local/cargo/bin/cargo test --workspace --features 'pg12 pg_test'"
+    - name: Run pgx tests
+      run: su postgres -c 'sh tools/build -pg$PGVERSION test-extension 2>&1'
 
+    # TODO post-install + doc test adds a good 90s.  We could probably put
+    # them together run serially in a separate job to shave that 90s off the
+    # long pole.
```

Comment:
`post-install` is testing update scripts, which should be safe to just do once. doc tests are mostly testing continuous aggregates, which I _think_ we want to do on every version 🤔